packages/cli/src/commands/hydrogen/deploy.test.ts

```diff
@@ -708,6 +728,82 @@ describe('deploy', () => {
     }
   });
 
+  it('throws error when authBypassTokenDuration is set but generateAuthBypassToken is false', async () => {
+    const params = {
+      ...deployParams,
+      generateAuthBypassToken: false,
+      authBypassTokenDuration: '3',
+    };
+
+    vi.mocked(createDeploy).mockImplementation(() => {
+      throw new AbortError(
+        'authBypassTokenDuration is set but generateAuthBypassToken is not enabled',
+      );
+    });
+
+    await expect(runDeploy(params)).rejects.toThrow(AbortError);
+    await expect(runDeploy(params)).rejects.toThrow(
+      'authBypassTokenDuration is set but generateAuthBypassToken is not enabled',
+    );
+  });
+
+  it('throws error if authBypassTokenDuration format is invalid', async () => {
+    const params = {
+      ...deployParams,
+      generateAuthBypassToken: true,
+      authBypassTokenDuration: '15h',
+    };
+
+    vi.mocked(createDeploy).mockImplementation(() => {
+      throw new AbortError(
+        "Invalid duration format. Please specify the duration in hours (e.g., '10').",
+      );
+    });
+
+    await expect(runDeploy(params)).rejects.toThrow(AbortError);
+    await expect(runDeploy(params)).rejects.toThrow(
+      "Invalid duration format. Please specify the duration in hours (e.g., '10').",
+    );
+  });
+
+  it('throws error if authBypassTokenDuration exceeds maximum allowed value', async () => {
+    const params = {
+      ...deployParams,
+      generateAuthBypassToken: true,
+      authBypassTokenDuration: '13',
+    };
+
+    vi.mocked(createDeploy).mockImplementation(() => {
+      throw new AbortError(
+        'Invalid duration length. Please specify the duration (in hours) between 1 to 12 hours.',
+      );
+    });
+
+    await expect(runDeploy(params)).rejects.toThrow(AbortError);
+    await expect(runDeploy(params)).rejects.toThrow(
+      'Invalid duration length. Please specify the duration (in hours) between 1 to 12 hours.',
+    );
+  });
+
+  it('throws error if authBypassTokenDuration is under minimum allowed value', async () => {
+    const params = {
+      ...deployParams,
+      generateAuthBypassToken: true,
+      authBypassTokenDuration: '0',
+    };
+
+    vi.mocked(createDeploy).mockImplementation(() => {
+      throw new AbortError(
+        'Invalid duration length. Please specify the duration (in hours) between 1 to 12 hours.',
+      );
+    });
+
+    await expect(runDeploy(params)).rejects.toThrow(AbortError);
+    await expect(runDeploy(params)).rejects.toThrow(
+      'Invalid duration length. Please specify the duration (in hours) between 1 to 12 hours.',
+    );
+  });
```

Comment:
Since the throw logic is entirely in oxygen-hub and not in this code, i dont think we need to test it here, but I'd love to hear other's opinions on this.

Reply:
👍🏻 we should only test what this repo is responsible for